File: packages/zynth-secure-store/ios/ZynthSecureStore.swift
//
//  ZynthSecureStore.swift
//  ZynthSecureStore
//
//  Public interface for ZynthSecureStore module
//

import Foundation
import ZynthKit

@objc(ZynthSecureStore)
public class ZynthSecureStore: NSObject {
  private static var moduleInstance: ZynthSecureStoreModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthSecureStore] Module already initialized")
      return
    }

    let module = ZynthSecureStoreModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthSecureStore] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/zynth-secure-store/android/ZynthSecureStore/src/main/java/dev/zynth/securestore/ZynthSecureStore.kt
package dev.zynth.securestore

import android.app.Activity
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthSecureStore module.
 */
object ZynthSecureStore {
    private var moduleInstance: ZynthSecureStoreModule? = null

    /**
     * Initialize the SecureStore module with an Activity and ZynthRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("ZynthSecureStore", "Module already initialized")
            return
        }

        moduleInstance = ZynthSecureStoreModule(activity)
        runtime.installModules(listOf(moduleInstance!!))
        android.util.Log.d("ZynthSecureStore", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}
